tests/helpers/tooling_contracts/mutation.py

```
@@ -0,0 +1,168 @@
+        # Split on operators before dequoting, so quoted text cannot forge a
+        # command boundary; comments are dropped by the same quote-aware scan.
+        for command in _split_shell_commands(line):
+            # Surface malformed shell instead of masking it as a valid install;
+            # a setup-commands script that cannot be parsed is not runnable.
+            try:
+                tokens = shlex.split(command)
+            except ValueError as error:
+                raise AssertionError(
+                    "expected mutation job setup-commands to be parseable "
+                    f"shell, got {command!r}"
+                ) from error
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)